snapshot/bazel/filer.go

```diff
@@ -1,44 +1,65 @@
 package bazel
 
 import (
+	"fmt"
+
+	"github.com/twitter/scoot/common/dialer"
+	"github.com/twitter/scoot/os/temp"
 	"github.com/twitter/scoot/snapshot"
 	"github.com/twitter/scoot/snapshot/snapshots"
 )
 
-func MakeBzFiler() *bzFiler {
-	return &bzFiler{
-		command: &bzCommand{},
-		updater: snapshots.MakeNoopUpdater(),
-	}
+func MakeBzFiler(tmp *temp.TempDir, r dialer.Resolver) (*BzFiler, error) {
+	return makeBzFiler(tmp, r, false)
 }
 
-// Options is a variadic list of functions that take a *bzCommand as an arg
-// and modify its fields, e.g.
-// localStorePath := func(bc *bzCommand) {
-//     bc.localStorePath = "/path/to/local/store"
-// }
-func MakeBzFilerWithOptions(options ...func(*bzCommand)) *bzFiler {
-	return &bzFiler{
-		updater: snapshots.MakeNoopUpdater(),
-		command: MakeBzCommandWithOptions(options...),
-	}
+func MakeBzFilerKeepCheckouts(tmp *temp.TempDir, r dialer.Resolver) (*BzFiler, error) {
+	return makeBzFiler(tmp, r, true)
 }
 
-func MakeBzFilerWithOptionsKeepCheckouts(options ...func(*bzCommand)) *bzFiler {
-	return &bzFiler{
+func makeBzFiler(tmp *temp.TempDir, r dialer.Resolver, keep bool) (*BzFiler, error) {
+	if tmp == nil {
+		return nil, fmt.Errorf("TempDir must be provided to MakeBzFiler")
+	}
+	runnerDir, err := tmp.TempDir("bzrunner")
+	if err != nil {
+		return nil, err
+	}
+
+	bf := &BzFiler{
+		runner:        makeBzCommand(runnerDir.Dir, r),
+		tmp:           tmp,
+		keepCheckouts: keep,
+		CASResolver:   r,
 		updater:       snapshots.MakeNoopUpdater(),
-		command:       MakeBzCommandWithOptions(options...),
-		keepCheckouts: true,
 	}
+	return bf, nil
 }
 
-// Satisfies snapshot.Checkouter, snapshot.Ingester, and snapshot.Updater
-// Default command is fs_util, a tool provided by github.com/pantsbuild/pants which
+// Implements Snapshot interface (snapshot.Checkouter, snapshot.Ingester, and snapshot.Updater)
+// Default runner is fs_util, a tool provided by github.com/pantsbuild/pants which
 // handles underlying implementation of bazel snapshot functionality
-type bzFiler struct {
-	command       bzRunner
-	updater       snapshot.Updater
+type BzFiler struct {
+	runner bzRunner
```

Comment:
A little late for this comment, but why is this considered a runner?

Reply:
runs underlying bazel operations... I'm open to other suggestions here, this would be easy to change. BzTreeer? BzFs? etc

Reply:
Seems confusing to reuse the term runner, anything else would be better as long as it's unique or thematically consistent.